### PR TITLE
Unify callback structures

### PR DESCRIPTION
**Pull request by esteve**
This branch removes the C++ callbacks structures and uses the plain C `message_type_support_callbacks_t` and `service_type_support_callbacks_t`

@dirk-thomas @tfoote @wjwwood ping

### CODE REVIEW

**Comment by tfoote** [comment]
lgtm

File: rmw_connext_cpp/src/functions.cpp [hunk 1]
```diff
@@ -25,16 +22,26 @@ rmw_get_implementation_identifier() {
...
-struct CustomServiceInfo {
+struct ConnextStaticSubscriberInfo {
+  DDSDataReader * topic_reader_;
+  const message_type_support_callbacks_t * callbacks_;
+};
+
+struct ConnextStaticPublisherInfo {
+  DDSDataWriter * topic_writer_;
+  const message_type_support_callbacks_t * callbacks_;
+};
```
Comment (dirk-thomas) [inline rmw_connext_cpp/src/functions.cpp]: Switch order of structs: PublisherInfo before SubscriberInfo.

Comment (esteve) [inline rmw_connext_cpp/src/functions.cpp]: Done. Also for client and service.

**Comment by wjwwood** [comment]
lgtm

**Comment by esteve** [comment]
Thanks for your reviews.

**Pull request merged**